src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.StandardTextDocument.cs

```diff
@@ -147,13 +147,23 @@ public TextLoader Loader
 
             public DocumentInfo GetInitialState()
             {
-                return DocumentInfo.Create(
+                var info = DocumentInfo.Create(
                     id: this.Id,
                     name: this.Name,
                     folders: this.Folders,
                     sourceCodeKind: this.SourceCodeKind,
                     loader: this.Loader,
                     filePath: this.FilePath);
+
+                var abstractProject = this.Project as AbstractProject;
+                if (abstractProject != null)
+                {
+                    // set project file path if possible. this will let us have persistable checksum
+                    // for the document
+                    info = info.WithProjectFilePath(abstractProject.ProjectFilePath);
```

Comment:
Err... this seems super wonky.  shoudl ProjectFilePath just move down to Project?  Or Project.Info/State?